featurebyte/service/feature_table_cache.py

```diff
@@ -527,32 +471,32 @@ async def create_or_update_feature_table_cache(
             remaining_progress_callback = None
 
         if non_cached_nodes:
-            if feature_table_cache_exists:
-                # if feature table cache exists - update existing table with new features
-                await self._update_table(
-                    feature_store=feature_store,
-                    observation_table=observation_table,
-                    cache_metadata=cache_metadata,
-                    db_session=db_session,
-                    graph=graph,
-                    non_cached_nodes=non_cached_nodes,
-                    is_target=is_target,
-                    serving_names_mapping=serving_names_mapping,
-                    progress_callback=remaining_progress_callback,
-                )
-            else:
-                # if feature table doesn't exist yet - create from scratch
-                await self._create_table(
-                    feature_store=feature_store,
-                    observation_table=observation_table,
-                    db_session=db_session,
-                    final_table_name=cache_metadata.table_name,
-                    graph=graph,
-                    nodes=non_cached_nodes,
-                    is_target=is_target,
-                    serving_names_mapping=serving_names_mapping,
-                    progress_callback=remaining_progress_callback,
+            if not feature_table_cache_exists:
+                # If cache table doesn't exist yet, create one by cloning from the observation table
+                request_column_names = [col.name for col in observation_table.columns_info]
+                await db_session.create_table_as(
+                    table_details=TableDetails(
+                        database_name=db_session.database_name,
+                        schema_name=db_session.schema_name,
+                        table_name=cache_metadata.table_name,
+                    ),
+                    select_expr=get_source_expr(
+                        observation_table.location.table_details,
+                        column_names=[InternalName.TABLE_ROW_INDEX.value] + request_column_names,
```

Comment:
Does the row index column always exist? Do we need to worry that user uploaded observation table does not contain the row index column?

Reply:
We can assume that row index column always exist since it's added by featurebyte service when creating observation table: https://github.com/featurebyte/featurebyte/blob/e1d7a79653a9133f5e44623b7571d689acd6bfa9/featurebyte/worker/task/observation_table.py#L55-L57

For very old observation tables without the row index column, they fallback to code paths without involving feature table cache: https://github.com/featurebyte/featurebyte/blob/e57a800dacc9532cef550672ab7ca1b81894a0fa/featurebyte/service/historical_features.py#L64-L68